core/src/main/java/org/mskcc/cbio/portal/servlet/CheckDarwinAccessServlet.java

```diff
@@ -104,7 +106,15 @@ public static String getResponse(String userName, String patientId){
             HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = getRequestEntity(userName, patientId);
             ResponseEntity<DarwinAccess> responseEntity = restTemplate.exchange(darwinAuthUrl, HttpMethod.POST, requestEntity, DarwinAccess.class);
             String darwinResponse = responseEntity.getBody().getDarwinAuthResponse();
-            return darwinResponse.equals("valid")?darwinResponseUrl+patientId:"";
+            String deidentificationId = responseEntity.getBody().getDeidentification_Id();
+            if (!darwinResponse.equals("valid")) {
+                return "";
+            }
+            if (deidentificationId.isEmpty()) {
+                return "";
+            }
+            // construct URL 
+            return ddpResponseUrl + "1597405" + DDP_INFO_ENDPOINT;
```

Comment:
@averyniceday  should this be here? those numbers..

Reply:
nope! whoops, I don't even know why I put that there. It's just supposed to be whatever deidentified pid we got from darwin's endpoint. Updated!